src/dispatch/Authentication.ts

```diff
@@ -4,16 +4,16 @@ import { Credentials } from '@aws-sdk/types';
 import { FetchHttpHandler } from '@aws-sdk/fetch-http-handler';
 import { StsClient } from './StsClient';
 import { CRED_KEY, CRED_RENEW_MS } from '../utils/constants';
+import { Configurable } from '../abstract/Configurable';
 
-export class Authentication {
+export class Authentication extends Configurable<Config> {
```

Comment:
issue: While the built-in plugins all share a `config` property, I don't see the benefit we get by extending other classes like this one.

Perhaps there is no downside either, but I think we should be mindful of over-using inheritance and generics, as they can make the code less understandable (i.e., more difficult to maintain and more likely to introduce bugs).

Reply:
Really we should not be passing this global "config" around that is shared and mutated by N number of classes...

Reply:
Right, so there is an increased risk that the web client or a plugin could modify the config.

I guess I'm not convinced that (1) this is a problem in practice, or that (2) the config should not be mutable. So adding a method call to access a config property seems unnecessary to me.